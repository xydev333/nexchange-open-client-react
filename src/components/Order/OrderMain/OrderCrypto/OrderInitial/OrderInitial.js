import React, { Component } from 'react';
import CopyToClipboard from 'react-copy-to-clipboard';
import styles from './OrderInitial.scss';

class OrderInitial extends Component {
  triggerCopyTooltip() {
    $('#copy-to-clipboard').tooltip({
      trigger: 'click',
      placement: 'top',
    });

    $('#copy-to-clipboard')
      .tooltip('hide')
      .attr('data-original-title', 'Wallet address copied!')
      .tooltip('show');

    setTimeout(() => {
      $('#copy-to-clipboard').tooltip('destroy');
    }, 1000);
  }

  getDepositAddressQr() {
    return `https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=
			${this.props.order.deposit_address.address}`;
  }

  render() {
    return (
      <div className={styles.container}>
        <div className={styles['qr-container']}>
          <img className={styles.qr} src={this.getDepositAddressQr()} alt="Deposit QR code" />
        </div>

        <div className={styles.details}>
          <h3>
            Time remaining:{' '}
            <span className={styles.time}>
              <b>{this.props.time}</b>
            </span>
          </h3>

          <h4>
            Send{' '}
            <b>
              {this.props.order.amount_quote} {this.props.order.pair.quote.code}
            </b>{' '}
            to the address<br />
            <b className={styles.address} style={{ wordWrap: 'break-word' }}>
              {this.props.order.deposit_address.address}
            </b>
          </h4>

          <CopyToClipboard text={this.props.order.deposit_address.address} onCopy={() => this.triggerCopyTooltip()}>
            <button id="copy-to-clipboard" type="button" className="btn btn-default">
              Copy address
            </button>
          </CopyToClipboard>
        </div>
      </div>
    );
  }
}

export default OrderInitial;
