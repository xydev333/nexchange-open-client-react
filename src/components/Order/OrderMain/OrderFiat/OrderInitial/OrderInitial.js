import React from 'react';

const OrderInitial = props => {
  return (
    <div id="order-payment" className="row">
      <div id="order-payment-details" className="col-xs-12 col-ms-6 col-sm-6 col-md-4">
        <h3>
          Time remaining:{' '}
          <span id="time-remaining">
            <b>{props.time}</b>
          </span>
        </h3>
        <h4>
          Pay{' '}
          <b>
            {parseFloat(props.order.amount_quote)} {props.order.pair.quote.code}
          </b>
        </h4>
        <a href={props.order.payment_url} target="_blank" className="btn btn-default btn-lg">
          <i className="fa fa-credit-card" aria-hidden="true" style={{ position: 'relative', left: -13 }} />
          Pay now
        </a>
      </div>

      <div id="cards" className="col-xs-12 col-ms-6 col-sm-6 col-md-8">
        <h3>Accepted cards:</h3>

        <div className="visible-xs-block visible-sm-block">
          <img src="/img/order/cards-mobile.png" alt="Credit cards accepted" />
        </div>

        <div className="visible-md-block visible-lg-block">
          <img src="/img/order/cards-desktop.png" alt="Credit cards accepted" />
        </div>
      </div>
    </div>
  );
};

export default OrderInitial;
