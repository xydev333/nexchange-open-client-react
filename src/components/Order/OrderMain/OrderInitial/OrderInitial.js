import React, { Component } from 'react';
import moment from 'moment';
import isFiatOrder from 'Utils/isFiatOrder';
import OrderInitialFiat from '../OrderFiat/OrderInitial/OrderInitial';
import OrderInitialCrypto from '../OrderCrypto/OrderInitial/OrderInitial';
import OrderExpired from '../OrderExpired/OrderExpired';

class OrderInitial extends Component {
  constructor(props) {
    super(props);
    this.state = { time: this.calculateRemainingTime() };

    this.calculateRemainingTime = this.calculateRemainingTime.bind(this);
    this.tick = this.tick.bind(this);
  }

  componentDidMount() {
    this.interval = setInterval(this.tick, 1000);
  }

  calculateRemainingTime() {
    const now = moment().subtract(this.props.order.payment_window, 'minutes');
    const createdOn = moment(this.props.order.created_on);
    let time = createdOn.diff(now);

    if (time < 0) {
      clearInterval(this.interval);
    } else {
      time = moment.utc(time).format('mm:ss');
    }

    return time;
  }

  tick() {
    if (!this.props.order) return;

    const time = this.calculateRemainingTime();
    this.setState({ time });
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    if (this.state.time < 0) {
      return <OrderExpired {...this.props} />;
    } else if (isFiatOrder(this.props.order)) {
      return <OrderInitialFiat time={this.state.time} {...this.props} />;
    } else {
      return <OrderInitialCrypto time={this.state.time} {...this.props} />;
    }
  }
}

export default OrderInitial;
