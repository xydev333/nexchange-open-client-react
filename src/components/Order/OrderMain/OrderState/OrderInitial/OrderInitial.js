import React, { Component } from 'react';
import moment from 'moment';
import axios from 'axios';
import config from 'Config';
import isFiatOrder from 'Utils/isFiatOrder';
import OrderInitialFiat from './OrderInitialFiat/OrderInitialFiat';
import OrderInitialCrypto from './OrderInitialCrypto/OrderInitialCrypto';
import OrderExpired from '../OrderExpired/OrderExpired';

class OrderInitial extends Component {
  constructor(props) {
    super(props);
    const time = this.calculateRemainingTime();
    this.state = { time, repeatOrder: false };
    this.tick = this.tick.bind(this);
    this.shouldRepeatOrder = false;
    this.alreadyRepeatedOrder = false;
    this.updatingRates = false;
  }

  UNSAFE_componentWillMount() {  
    if(!(this.state.time < 0)) {
      this.interval = setInterval(this.tick, 1000);
      this.shouldRepeatOrder = true;
      this.alreadyRepeatedOrder = false;
      this.updatingRates = false;
    }
  }

  UNSAFE_componentWillReceiveProps = nextProps => {
    //New Order
    if(this.props.order.unique_reference !== nextProps.order.unique_reference) {
      const time = this.calculateRemainingTime(nextProps.order);   
      this.setState({time, repeatOrder: false});
    }

    //Rates are updated, enable repeat order
    if (!this.state.repeatOrder && !this.alreadyRepeatedOrder && this.updatingRates) {
      this.alreadyRepeatedOrder = true;
      this.updatingRates = false;
      this.setState({repeatOrder: true});
      }
    }

  UNSAFE_componentWillUpdate(nextProps, nextState) {
    //Fetch updated rates for the expired order
    if (nextState.time < 0 && this.shouldRepeatOrder) { 
      const order = this.props.order;

      const data = {
        pair: order.pair.name,
        lastEdited: 'deposit',
      };

      data['deposit'] = order.amount_quote;
      this.shouldRepeatOrder = false;
      this.props.fetchPrice(data);
      this.updatingRates = true;
      clearInterval(this.interval);
    } else if (!(nextState.time < 0) && !this.shouldRepeatOrder) {
      this.shouldRepeatOrder = true;
      this.alreadyRepeatedOrder = false;
      this.interval = setInterval(this.tick, 1000);
    }

    //Repeat the order at updated rates
    if(nextState.repeatOrder) {
      this.setState({repeatOrder: false});
      const order = nextProps.order;
      const price = nextProps.price;
      const data = {
        amount_base: price.receive,
        amount_quote: order.amount_quote,
        is_default_rule: true,
        pair: {
          name: order.pair.name,
        },
        withdraw_address: {
          address: order.withdraw_address.address,
          name: '',
          payment_id: order.withdraw_address.payment_id,
          destination_tag: order.withdraw_address.destination_tag,
          memo: order.withdraw_address.memo,
        },
      };
  
      axios
      .post(`${config.API_BASE_URL}/orders/`, data)
      .then(response => {
        const data = response.data;
        this.props.setOrder(data);
        this.props.history.push(`/order/${data.unique_reference}`)
      })
      .catch(error => {
        console.log('Error:', error);
      });
    }
  }


  calculateRemainingTime = (nextPropsOrder) => {
    const order = nextPropsOrder || this.props.order;
    const now = moment()
      .utc()
      .subtract(order.payment_window, 'minutes');
    const createdOn = moment(order.created_on);
    let time = createdOn.diff(now);

    if (time < 0) {
      clearInterval(this.interval);
    } else {
      time = moment.utc(time).format('mm:ss');
    }

    return time;
  };

  tick() {
    const time = this.calculateRemainingTime();    
    this.setState({ time });
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    if (!this.props.isLimitOrder && this.state.time < 0) {
      return <OrderExpired {...this.props} />;
    } else if (isFiatOrder(this.props.order)) {
      return <OrderInitialFiat time={this.state.time} {...this.props} />;
    } else {
      return <OrderInitialCrypto time={this.state.time} {...this.props} />;
    }
  }
}

export default OrderInitial;
