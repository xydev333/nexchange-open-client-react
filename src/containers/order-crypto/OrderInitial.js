import React, { Component } from 'react';
import {Icon} from 'react-fa';
import CopyToClipboard from 'react-copy-to-clipboard';


class OrderInitial extends Component {
	constructor(props) {
		super(props);
	}

	triggerCopyTooltip() {
		$('#copy-to-clipboard').tooltip({
			trigger: 'click',
			placement: 'top'
		});

		$('#copy-to-clipboard').tooltip('hide')
			.attr('data-original-title', 'Wallet address copied!')
			.tooltip('show');

		setTimeout(() => {
			$('#copy-to-clipboard').tooltip('destroy');
		}, 1000);
	}
	
	getDepositAddressQr () {
		return "https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=" 
			+ this.props.depositCoinName + ":" 
			+ this.props.depositAddress + "?amount=" 
			+ this.props.depositAmount;
	}

	render() {
	    return (
	    	<div id="order-payment">
    			<div className="col-xs-12 col-sm-4 col-md-3">
    				<img src={this.getDepositAddressQr()} />
    			</div>

    			<div id="order-payment-details" className="col-xs-12 col-sm-8 col-md-9">
    				<h3>Time remaining: <span id="time-remaining"><b>{this.props.timeRemaining}</b></span></h3>

    				<h4>Send <b>{this.props.order.amount_quote} {this.props.order.pair.quote.code}</b> to the address<br/>
    					<b id="deposit-address">{this.props.order.deposit_address.address}</b>
    				</h4>
    				

			        <CopyToClipboard text={this.props.order.deposit_address.address} onCopy={() => this.triggerCopyTooltip()}>
						<button id="copy-to-clipboard" type="button" className="btn btn-default btn-themed">Copy the address</button>
    				</CopyToClipboard>
    			</div>
		    </div>
	    );
	}
}

export default OrderInitial;
