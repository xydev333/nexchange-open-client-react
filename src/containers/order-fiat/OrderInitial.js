import React, { Component } from 'react';
import axios from 'axios';
import config from '../../config';
import KYCModal from './KYCModal';
import DesktopNotifications from '../DesktopNotifications';
import OrderPaymentForm from './OrderPaymentForm';

class OrderInitial extends Component {
	constructor(props) {
		super(props);
		this.state = { showKYCModal: false, ordersCount: 1 }
		this.checkKYC = this.checkKYC.bind(this);
	}

	componentDidMount() {
		this.checkKYC(true);
	
		axios
			.get(`${config.API_BASE_URL}/users/me/orders/`)
			.then(response => {
				this.setState({ ordersCount: response.data.count })
			});
	}

	checkKYC(firstTime) {
		clearTimeout(this.timeout);

		axios
			.get(`${config.API_BASE_URL}/kyc/${this.props.order.unique_reference}`)
			.then(response => {
				const kyc = response.data;
				this.setState({ kyc });

				if (firstTime && (!kyc.id_document_status || !kyc.residence_document_status)) {
					setTimeout(() => {
						this.setState({ showKYCModal: true });
					}, 1000);
				}

				this.timeout = setTimeout(() => {
					this.checkKYC();
				}, config.KYC_DETAILS_FETCH_INTERVAL);
			})
			.catch(error => {
				this.timeout = setTimeout(() => {
					this.checkKYC();
				}, config.KYC_DETAILS_FETCH_INTERVAL);
			});
	}

	componentWillUnmount() {
		clearTimeout(this.timeout);
	}

	render() {
		let inner;
		let buttonText;
		let notificationsCtaVisible = false;

		if (!this.state.kyc) {
			inner = <h2>Checking KYC status...</h2>;
		} else if (this.state.kyc.id_document_status !== 'APPROVED' && this.state.kyc.residence_document_status !== 'APPROVED') {
			if (this.state.kyc.id_document_status === 'UNDEFINED' && this.state.kyc.residence_document_status === 'UNDEFINED') {
				inner = (
					<div>
						<h2>Awaiting verification</h2>
						<h5>In order to proceed further we must get to know you better by getting a copy of your government issued ID and a proof of residence.</h5>
						<h5 style={{marginTop: 15}}>
							<b>This is a one-time process, once verified youâ€™ll be able to complete future purchases instantly.</b>
						</h5>
					</div>
				);

				buttonText = "Get verified";
			} else {
				inner = (
					<div>
						<h2>Verification received, awaiting approval</h2>

						<hr style={{marginLeft: -15, marginRight: -15}} />

						<h2>Approval status:</h2>
						<p style={{margin:0}}><b>Government issued ID:</b> {this.state.kyc.id_document_status}</p>
						<p><b>Proof of residence:</b> {this.state.kyc.residence_document_status}</p>
						
						{this.state.kyc && this.state.kyc.user_visible_comment &&
							<p><b>Reason for rejection: {this.state.kyc.user_visible_comment}</b></p>}
					</div>
				);

				notificationsCtaVisible = true;

				if (this.state.kyc.id_document_status === 'REJECTED' && this.state.kyc.residence_document_status === 'REJECTED') {
					buttonText = "Retry verification";
				} else if (
					(this.state.kyc.id_document_status === 'APPROVED'
					|| this.state.kyc.id_document_status === 'PENDING')
					&& this.state.kyc.residence_document_status === 'REJECTED') {
					buttonText = "Retry proof of residence verification";
				} else if (
					(this.state.kyc.residence_document_status === 'APPROVED'
					|| this.state.kyc.residence_document_status === 'PENDING')
					&& this.state.kyc.id_document_status === 'REJECTED') {
					buttonText = "Retry government issued ID verification";
				}
			}
		} else if (this.state.kyc.id_document_status === 'APPROVED' && this.state.kyc.residence_document_status === 'APPROVED') {
			return <OrderPaymentForm {...this.props} />;
		}

		return <div className="col-xs-12 text-center order-status-section">
			{inner}

			<DesktopNotifications kyc={this.state.kyc} {...this.props} visible={notificationsCtaVisible} />

			{buttonText &&
				<button
					type="button"
					className="btn btn-default btn-themed"
					onClick={() => this.setState({showKYCModal: true})}
					style={{marginTop: 20}}>
					<i className="fa fa-credit-card" aria-hidden="true" style={{position: "relative", left: -13}}></i>
					{buttonText}
				</button>}

			{this.state.kyc &&
				<KYCModal
					show={ this.state.showKYCModal } 
					onClose={() => {
						this.setState({ showKYCModal: false });
						this.checkKYC();
					}}
					kyc={this.state.kyc}
					{...this.props} />}
		</div>
	}

};

export default OrderInitial;
