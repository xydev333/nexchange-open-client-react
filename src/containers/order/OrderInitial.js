import React, { Component } from 'react';
import moment from 'moment';
import Helpers from '../../helpers';

import OrderInitialFiat from '../order-fiat/OrderInitial';
import OrderInitialCrypto from '../order-crypto/OrderInitial';
import OrderExpired from '../order/OrderExpired';
import Loading from '../../components/Loading';

class OrderInitial extends Component {
    constructor(props) {
        super(props);
        this.state = { time: '...' }
        this.tick = this.tick.bind(this);
    }

    componentDidMount() {
        this.interval = setInterval(this.tick, 1000);
    }

	tick() {
		if (!this.props.order) return;

		const now = moment().subtract(this.props.order.payment_window, 'minutes');
		const createdOn = moment(this.props.order.created_on);
		let time = createdOn.diff(now);

		if (time < 0) {
			clearInterval(this.interval);
		} else {
			time = moment.utc(time).format('mm:ss')
		}

		this.setState({ time });
	}

    componentWillUnmount() {
        clearInterval(this.interval);
    }

    render() {
        if (this.state.time === '...') {
            return <Loading />
        } else if (this.state.time < 0) {
            return <OrderExpired {...this.props} />;
        } else if (Helpers.isFiatOrder(this.props.order)) {
            return <OrderInitialFiat time={this.state.time} {...this.props} />
        } else {
            return <OrderInitialCrypto time={this.state.time} {...this.props} />
        }
    }
}

export default OrderInitial;