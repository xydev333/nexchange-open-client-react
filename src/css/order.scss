@import "variables";
@import "mixins";
@import "bootstrap/variables";
@import "bootstrap/mixins";

#order { background: #f8f7f8; padding: 15px 0 0 0;
	a { color: $green; 
		&:visited, &:focus { text-decoration: none; }
		&:hover { text-decoration: underline; }
		&.btn { color: #fff;
			&:hover { text-decoration: none; }
		}
	}
	#order-ref { margin-bottom: 15px; }
	.box {
		h4 { font-size: 15px; }
		h5, h6 { margin: 0; text-transform: none; }
		.coin-icon { position: relative; top: 8px; font-size: 28px;
			@media (min-width: $screen-sm-min) { top: 9px; }
			@media (min-width: $screen-md-min) { top: 11px; }
		}
		.text-warning { color: $green; }		

		@media (min-width: $screen-sm-min) { margin-bottom: 30px; }
	}

	&.order-fiat {
		.box.fiat {
			h5 { margin-top: 11px; padding-bottom: 7px;
				@media (min-width: $screen-sm-min) { margin-top: 11px; padding-bottom: 13px; }
			}
		}
	}

	.coin-box { padding-left: 10px;
		.tooltip-inner { text-align: left; }
	}
	#time-remaining { color: #2cb4a0; }
	.order-status-section { 
		h2 { margin-top: 0; }
		.explanation { font-size: 16px; margin-top: 15px; }
	}


	@media (min-width: $screen-sm-min) { padding: 25px 0;
		h5 { font-size: 15px; }
		h6 { font-size: 12px; }
		.box h4 { font-size: 18px; }
	}

	@media (min-width: $screen-md-min) {
		h5 { font-size: 17px; }
		h6 { font-size: 16px; }
	}
}

#order-header {
	#bookmark-button { font-size: 20px; padding: 0; margin-top: 0; color: $green; }

	@media (min-width: $screen-ms-min) {
		h3 { float: left; }
		#bookmark-button { float: right; margin-top: 22px; }
	}
}

#order-payment { text-align: center;
	h3 { margin-top: 0; }
	@media (min-width: $screen-ms-min) { text-align: left; }
}

#bookmark {
	.bookmark-text{ font-size: 19px; }
	.input-group-addon { cursor: pointer; padding-top: 0;
		i { font-size: 18px; }
	}
	.form-group { margin-top: 10px; }
	input { font-size: 18px; }
}

.order-status-section {
	h2 { font-size: 24px; }

	@media (min-width: $screen-sm-min) {
		h2 { font-size: 30px; }
	}
}

#share-referral, #notifications { text-align: center;
	.box { padding: 20px; }
	.share { margin-top: 15px; }
	.fa { font-size: 45px; margin: 0 15px 0 0; color: #d6d6d6; }
	h2 { margin-top: 0; font-size: 22px;
		@media (min-width: $screen-sm-min) { font-size: 26px; }
		@media (min-width: $screen-md-min) { font-size: 34px; }
	}
}

#notifications {
	.form-group { margin-top: 10px; }
	button { width: 100%; margin-bottom: 0; }
	form { padding-top: 15px;
		h4 { position: absolute; top: -15px; left: 50%; transform: translate(-50%, 0); }
	}
	@media (min-width: $screen-sm-min) { button { width: auto; } }
}

@include keyframes(color-change) {
	from { color: #adadad; }
	to { color: #fbc02d; } 
}

@import "order-status";
@import "referral";
