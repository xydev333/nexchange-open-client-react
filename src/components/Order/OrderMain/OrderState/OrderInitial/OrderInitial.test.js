import React from 'react';
import { shallow } from 'enzyme';
import MockDate from 'mockdate';
import OrderInitial from './OrderInitial';
import order from 'Mocks/order';
import orderFiat from 'Mocks/orderFiat';

describe('OrderInitial', () => {
  let wrapShallowFiat, wrapShallowCrypto;

  beforeEach(() => {
    MockDate.set(1531839850000);
    order.created_on = '2018-07-17T18:03:33.793300+03:00';

    wrapShallowCrypto = shallow(<OrderInitial order={order} />);
    wrapShallowFiat = shallow(<OrderInitial order={orderFiat} />);
  });

  it('renders correctly', () => {
    expect(wrapShallowCrypto).toMatchSnapshot();
    expect(wrapShallowFiat).toMatchSnapshot();
  });
});
