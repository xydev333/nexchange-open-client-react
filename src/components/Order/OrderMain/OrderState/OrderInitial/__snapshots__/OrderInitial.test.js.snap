// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInitial renders correctly 1`] = `ShallowWrapper {}`;

exports[`OrderInitial renders correctly 2`] = `ShallowWrapper {}`;

exports[`OrderInitial renders correctly 3`] = `ShallowWrapper {}`;
